appinventor/appengine/src/com/google/appinventor/server/project/youngandroid/YoungAndroidSettingsBuilder.java

```diff
@@ -0,0 +1,292 @@
+    this.versionName = versionName;
+    return this;
+  }
+
+  public YoungAndroidSettingsBuilder setUsesLocation(String usesLocation) {
+    this.usesLocation = usesLocation;
+    return this;
+  }
+
+  public YoungAndroidSettingsBuilder setAppName(String appName) {
+    this.appName = appName;
+    return this;
+  }
+
+  public YoungAndroidSettingsBuilder setSizing(String sizing) {
+    this.sizing = sizing;
+    return this;
+  }
+
+  public YoungAndroidSettingsBuilder setShowListsAsJson(String showListsAsJson) {
+    this.showListsAsJson = showListsAsJson;
+    return this;
+  }
+
+  public YoungAndroidSettingsBuilder setTutorialUrl(String tutorialUrl) {
+    this.tutorialUrl = tutorialUrl;
+    return this;
+  }
+
+  public YoungAndroidSettingsBuilder setBlocksSubset(String blockSubset) {
+    this.blockSubset = blockSubset;
+    return this;
+  }
+
+  public YoungAndroidSettingsBuilder setActionBar(String actionBar) {
+    this.actionBar = actionBar;
+    return this;
+  }
+
+  public YoungAndroidSettingsBuilder setTheme(String theme) {
+    this.theme = theme;
+    return this;
+  }
+
+  public YoungAndroidSettingsBuilder setPrimaryColor(String primaryColor) {
+    this.primaryColor = primaryColor;
+    return this;
+  }
+
+  public YoungAndroidSettingsBuilder setPrimaryColorDark(String primaryColorDark) {
+    this.primaryColorDark = primaryColorDark;
+    return this;
+  }
+
+  public YoungAndroidSettingsBuilder setAccentColor(String accentColor) {
+    this.accentColor = accentColor;
+    return this;
+  }
+
+  /**
+   * Convert the internal settings into a JSON structure.
+   *
+   * @return a JSON string containing the settings
+   */
+  public String build() {
+    JSONObject object = new JSONObject();
+    object.put(YOUNG_ANDROID_SETTINGS_ICON, icon);
+    object.put(YOUNG_ANDROID_SETTINGS_VERSION_CODE, versionCode);
+    object.put(YOUNG_ANDROID_SETTINGS_VERSION_NAME, versionName);
+    object.put(YOUNG_ANDROID_SETTINGS_USES_LOCATION, usesLocation);
+    object.put(YOUNG_ANDROID_SETTINGS_APP_NAME, appName);
+    object.put(YOUNG_ANDROID_SETTINGS_SIZING, sizing);
+    object.put(YOUNG_ANDROID_SETTINGS_SHOW_LISTS_AS_JSON, showListsAsJson);
+    object.put(YOUNG_ANDROID_SETTINGS_TUTORIAL_URL, tutorialUrl);
+    object.put(YOUNG_ANDROID_SETTINGS_BLOCK_SUBSET, blockSubset);
+    object.put(YOUNG_ANDROID_SETTINGS_ACTIONBAR, actionBar);
+    object.put(YOUNG_ANDROID_SETTINGS_THEME, theme);
+    object.put(YOUNG_ANDROID_SETTINGS_PRIMARY_COLOR, primaryColor);
+    object.put(YOUNG_ANDROID_SETTINGS_PRIMARY_COLOR_DARK, primaryColorDark);
+    object.put(YOUNG_ANDROID_SETTINGS_ACCENT_COLOR, accentColor);
+    JSONObject wrapper = new JSONObject();
+    wrapper.put(PROJECT_YOUNG_ANDROID_SETTINGS, object);
+    return wrapper.toString();
+  }
+
+  /**
+   * Convert the internal settings into a Java Properties object.
+   *
+   * @return a properties file contents containing the settings
+   */
+  public String toProperties() {
```

Comment:
Maybe I am overlooking something, but I don't see `aname` prop here

Reply:
Yes, I'm not sure how I missed that key when I was translating the code over.